### DISCUSSION TITLE
Bezier Curve Editor Suggestions

### DISCUSSION THREAD

**Discussion by AxZelAnims** [godotengine/godot-proposals#8389] [General Discussions]
Whenever in AnimationPlayer's Main Timeline or Bezier Curve Timeline. Adding a new Keyframe, there should be an option to choose what kind of handles will be used. 

The problem is that everytime you insert a keyframe, it automatically sets it up as Handles Free and you are forced to keep changing between Main or Bezier Curve Timeline. Using Linear Handles are very convenient when making basic keyframe testing, and having the option to insert a new keyframe with Linear Handle makes for easier keyframe adjustments. Bezier Curve Timeline can then be used for cleanup after all Linear Handle Curves are placed. Thus making Animation in AnimationPlayer alot easier. 

Additionally, adjust the Bezier Curve Editor's Handles because they are overlapping each other and it's harder to click between the keyframes/curve handles. 

AnimationPlayer and Bezier Curve Editor should also have a Vertical Zoom.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/6681.

> Additionally, adjust the Bezier Curve Editor's Handles because they are overlapping each other and it's harder to click between the keyframes/curve handles.

On the other hand, making their clickable area smaller will make them more difficult to quickly select (for reasons similar to what I mentioned in https://github.com/godotengine/godot-proposals/issues/8264#issuecomment-1786566547).

> AnimationPlayer and Bezier Curve Editor should also have a Vertical Zoom.

This is already being tracked in https://github.com/godotengine/godot-proposals/issues/7547.
